tests/benchmarks/thread_metric/scripts/twister_postprocess.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This script post-process Twister report (`twister.json`) after the Thread-Metric
+benchmark execution: each test suite of tests/benchmarks/thread_metric
+(benchmark.thread_metric.*) will have additional recording properties:
+
+  `benchmark_name` - the corresponding Thread-Metric benchmark name.
+  `baseline_median` - baseline metric median used to normalize other values.
+  `total_time_period_median` - current metric's median value.
+  `total_time_period_normalized` - current metric's values normalized - divided
+                                   by the baseline median value.
+
+These additional properties are to simplify further data analysis, for example
+to visualize the thread_metrics data collected on ElasticSearch Kibana dashboards
+using standard widgets.
+"""
+
+import argparse
+import json
+import os
+import statistics
+import sys
+
+has_errors = 0
+threadx_benchmark_name_prefix = "benchmark.thread_metric"
+threadx_benchmark_name = {
+    "basic": "Basic Processing (baseline)",
+    "cooperative": "Cooperative Scheduling",
+    "preemptive": "Preemptive Scheduling",
+    "interrupt": "Interrupt Processing",
+    "interrupt_preemption": "Interrupt Preemption Processing",
+    "message": "Message Processing",
+    "synchronization": "Synchronization Processing",
+    "memory_allocation": "RTOS memory allocation",
+}
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        allow_abbrev=False, formatter_class=argparse.RawTextHelpFormatter, description=__doc__
+    )
+    parser.add_argument(
+        'files',
+        metavar='TWISTER_JSON_FILE',
+        nargs='+',
+        help='twister.json report file to update with additional attributes.',
+    )
+    args = parser.parse_args()
+    return args
+
+
+def preprocess(json_report, fname):
+    global has_errors
```

Comment:
Maybe make this a local variable and return it alongside `report` to avoid a global and make the code easier to follow, debug and maybe extend?